packages/bun/src/integrations/bunserver.ts

```diff
@@ -53,89 +63,228 @@ export function instrumentBunServe(): void {
+    },
+  });
+}
 
-        const parsedUrl = parseUrl(request.url);
-        const attributes: SpanAttributes = {
-          [SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN]: 'auto.http.bun.serve',
-          [SEMANTIC_ATTRIBUTE_HTTP_REQUEST_METHOD]: request.method || 'GET',
-          [SEMANTIC_ATTRIBUTE_SENTRY_SOURCE]: 'url',
-        };
-        if (parsedUrl.search) {
-          attributes['http.query'] = parsedUrl.search;
-        }
+/**
+ * Instruments the `routes` option of Bun.serve.
+ *
+ * @param serveOptions - The options for the Bun.serve function.
+ */
+function instrumentBunServeOptionRoutes(serveOptions: Parameters<typeof Bun.serve>[0]): void {
+  if (!serveOptions.routes) {
+    return;
+  }
 
-        const url = getSanitizedUrlString(parsedUrl);
-
-        isolationScope.setSDKProcessingMetadata({
-          normalizedRequest: {
-            url,
-            method: request.method,
-            headers: request.headers.toJSON(),
-            query_string: extractQueryParamsFromUrl(url),
-          } satisfies RequestEventData,
-        });
-
-        return continueTrace(
-          { sentryTrace: request.headers.get('sentry-trace') || '', baggage: request.headers.get('baggage') },
-          () => {
-            return startSpan(
-              {
-                attributes,
-                op: 'http.server',
-                name: `${request.method} ${parsedUrl.path || '/'}`,
-              },
-              async span => {
-                try {
-                  const response = await (fetchTarget.apply(fetchThisArg, fetchArgs) as ReturnType<
-                    typeof serveOptions.fetch
-                  >);
-                  if (response?.status) {
-                    setHttpStatus(span, response.status);
-                    isolationScope.setContext('response', {
-                      headers: response.headers.toJSON(),
-                      status_code: response.status,
-                    });
-                  }
-                  return response;
-                } catch (e) {
-                  captureException(e, {
-                    mechanism: {
-                      type: 'bun',
-                      handled: false,
-                      data: {
-                        function: 'serve',
-                      },
-                    },
-                  });
-                  throw e;
-                }
+  if (typeof serveOptions.routes !== 'object') {
+    return;
+  }
+
+  Object.keys(serveOptions.routes).forEach(route => {
+    const routeHandler = serveOptions.routes[route];
+    if (typeof routeHandler === 'function') {
+      serveOptions.routes[route] = new Proxy(routeHandler, {
+        apply: (routeHandlerTarget, routeHandlerThisArg, routeHandlerArgs: Parameters<typeof routeHandler>) => {
+          return wrapRequestHandler(routeHandlerTarget, routeHandlerThisArg, routeHandlerArgs, route);
+        },
+      });
+    }
+
+    // TODO: Handle static routes
```

Comment:
l: IIUC correctly, we likely won't ever instrument static routes to avoid breaking the customization? So is this TODO even necessary?

```suggestion
```